src/Tools/dotnet-monitor/Exceptions/ExceptionsService.cs

```diff
@@ -66,8 +69,11 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
                         throw new NotSupportedException();
                     }
 
+                    bool isStartupHookApplied = false;
+                    _ = _startupHookEndpointInfoSourceCallbacks.ApplyStartupState.TryGetValue(pi.EndpointInfo.RuntimeInstanceCookie, out isStartupHookApplied);
+
                     // Validate that the process is configured correctly for collecting exceptions.
-                    if (!await _startupHookValidator.CheckAsync(pi.EndpointInfo, stoppingToken))
+                    if (!isStartupHookApplied && !await _startupHookValidator.CheckEnvironmentAsync(pi.EndpointInfo, stoppingToken))
```

Comment:
Because `_startupHookEndpointInfoSourceCallbacks.ApplyStartupState` is tracking the startup hook state, the `CheckEnvironmentAsync` call and the `_unconfiguredProcesses` list can be removed.